dev-lang/rust/rust-1.74.1-r100.ebuild

```
@@ -504,7 +530,257 @@ src_configure() {
 	echo
 }
 
+# Build a very minimal llvm that we can use for bootstrap rustc codegen
+llvm_bootstrap() {
+	# Reference ${P}/src/bootstrap/native.rs for these values
+	local llvm_cmake_opts=(
+		"-G Ninja"
+		"-DLLVM_TARGET_ARCH=${CFG_COMPILER_HOST_TRIPLE%%-*}"
+		"-DLLVM_DEFAULT_TARGET_TRIPLE=${CFG_COMPILER_HOST_TRIPLE}"
+		"-DLLVM_TARGETS_TO_BUILD=${BOOTSTRAP_LLVM_TARGETS:=X86;ARM;AArch64}" #;Mips;PowerPC;SystemZ;JSBackend;MSP430;Sparc;NVPTX
+		"-DLLVM_ENABLE_ASSERTIONS=OFF"
+		"-DLLVM_INCLUDE_EXAMPLES=OFF"
+		"-DLLVM_INCLUDE_TESTS=OFF"
+		"-DLLVM_INCLUDE_DOCS=OFF"
+		"-DLLVM_INCLUDE_BENCHMARKS=OFF"
+		"-DLLVM_ENABLE_ZLIB=OFF"
+		"-DLLVM_ENABLE_TERMINFO=OFF"
+		"-DLLVM_ENABLE_LIBEDIT=OFF"
+		"-DCMAKE_CXX_COMPILER=$(tc-getCXX)"
+		"-DCMAKE_C_COMPILER=$(tc-getCC)"
+		"-DCMAKE_BUILD_TYPE=Release"
+	)
+
+	if [[ -z "${LLVM_CMAKE_OPTS_EXTRA}" ]]; then
+		llvm_cmake_opts+=( "${LLVM_CMAKE_OPTS_EXTRA}")
+	fi
+
+	elog "Building bootstrap llvm"
+
+	mkdir -p "${WORKDIR}/bootstrap/llvm" || die
+	pushd "${WORKDIR}/bootstrap/llvm" 2>/dev/null || die
+		cmake "${llvm_cmake_opts[@]}" "${S}/src/llvm-project/llvm" || die "Failed to configure bootstrap llvm"
```

I believe this is a great place to use `edo` from `edo.eclass`, so we also log the full llvm invokation

```suggestion
		edo cmake "${llvm_cmake_opts[@]}" "${S}/src/llvm-project/llvm"
```